src/server/master_server/chunk_server_manager.cc

```diff
@@ -0,0 +1,62 @@
+#include "chunk_server_manager.h"
+
+namespace gfs {
+
+// Iterates through the chunk servers and selects the server with
+// the max available memory.
+std::shared_ptr<ChunkServer> ChunkServerManager::GetAvailableChunkServer() {
```

Comment:
This will happen a lot, and looping is expensive. Put a TODO comment for optimizations in future?

Reply:
yeah, this is first phase and a simple implementation. Added TODO.